.github/actions/setup-nix/action.yaml

```diff
@@ -20,14 +20,19 @@ runs:
     - name: Restore and cache Nix store
       uses: nix-community/cache-nix-action@v4.0.3
       with:
-        key: cache-nix-${{ runner.os }}-id-${{ inputs.cache-id }}-${{ hashFiles('nix/**/*.nix') }}
+        key: cache-nix-${{ runner.os }}-id-${{ inputs.cache-id }}-${{ hashFiles('nix/**/*.nix', '.github/actions/setup-nix/*') }}
         restore-keys: |
-          cache-nix-${{ runner.os }}-common-
-        restore-key-hit: true
+          cache-nix-${{ runner.os }}-id-
     - uses: cachix/cachix-action@v12
       with:
         name: postgrest
         authToken: ${{ inputs.authToken }}
+    - name: Put all tools to store to be cached afterwards
+      run: |
+        echo "::group::Downloading tools"
+        nix-store -v --realize $( nix-instantiate default.nix )
+        echo "::endgroup::"
+      shell: bash
```

Comment:
This is executed for every job that uses the setup-nix action. Since it's only needed to generate the first cache, maybe it should be moved to the Prepopulate job instead?

Reply:
Good point! Will do